### PR TITLE
 feat(autofix): Repo access & indexing steps + loading state for setup

### PR DESCRIPTION
**Pull request by jennmueng**
Adds a repo check & codebase indexing start step to the Autofix setup modal

Screenshot 2024-04-30 at 10 32 45 AM
Screenshot 2024-05-01 at 1 19 31 PM

Also adds a loading state to the autofix banner when a codebase is indexing
Screenshot 2024-05-01 at 1 19 37 PM

### CODE REVIEW

File: static/app/components/modals/autofixSetupModal.tsx [hunk 1]
```diff
@@ -197,3 +371,34 @@ const DoneIcon = styled(IconCheckmark)`
...
+
+const RepoLinkItem = styled('li')`
+  display: flex;
+  align-items: center;
+  gap: ${space(0.5)};
+`;
+
+const RepoCheckIcon = styled(IconCheckmark)`
+  fill: ${p => p.theme.success};
+`;
```
Comment (scttcper) [inline static/app/components/modals/autofixSetupModal.tsx:389]: icons take a color prop

Comment (scttcper) [reply static/app/components/modals/autofixSetupModal.tsx:389]: https://github.com/getsentry/sentry/blob/master/static/app/components/reprocessedBox.tsx#L81

File: static/app/components/events/autofix/useAutofixCodebaseIndexing.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  const startIndexing = useCallback(() => {
+    if (!projectSlug) {
+      return;
+    }
+
+    // Triggering the create endpoint on the seer side should make it start return indexing status...
+    api.requestPromise(makeCodebaseIndexCreateUrl(organization.slug, projectSlug), {
+      method: 'POST',
+    });
```
Comment (scttcper) [inline static/app/components/events/autofix/useAutofixCodebaseIndexing.tsx:73]: its a bit funky this promise is created and not caught or awaited but if you don't need to then go for it

**Review by scttcper — APPROVED** [review]

File: static/app/components/events/autofix/useAutofixCodebaseIndexing.tsx (hunk 2 above)
Comment (jennmueng) [reply static/app/components/events/autofix/useAutofixCodebaseIndexing.tsx:73]: yeah that's true, but basically we want to have instant feedback on the other query key that it's indexing

**Pull request merged**